superwall/src/main/java/com/superwall/sdk/paywall/presentation/internal/operators/GetPresenter.kt

```diff
@@ -94,13 +95,34 @@ suspend fun Superwall.getPresenterIfNecessary(
     return currentActivity
 }
 
-private suspend fun Superwall.activateSession(
+suspend fun Superwall.attemptTriggerFire(
     request: PresentationRequest,
     triggerResult: InternalTriggerResult,
-): String? {
+) {
+    if (request.flags.type == PresentationRequestType.GetImplicitPresentationResult ||
+        request.flags.type == PresentationRequestType.GetPresentationResult
+    ) {
+        return
+    }
     val sessionEventsManager = dependencyContainer.sessionEventsManager
-    return sessionEventsManager?.triggerSession?.activateSession(
-        request.presentationInfo,
-        triggerResult,
-    )
+    // If eventName is null, the paywall is being presented by identifier, which is what the debugger uses and that's not supported.
+    val eventName = request.presentationInfo.eventName ?: return
+
+    when (val req = request.presentationInfo) {
+        is PresentationInfo.ExplicitTrigger, is PresentationInfo.ImplicitTrigger -> {
+            when (triggerResult) {
+                is InternalTriggerResult.Error, is InternalTriggerResult.EventNotFound ->
+                    return
+                else -> {} // No-op
+            }
+        }
+        is PresentationInfo.FromIdentifier -> {} // No-op
+    }
+    val trackedEvent =
+        InternalSuperwallEvent.TriggerFire(
+            triggerResult = triggerResult,
+            triggerName = eventName,
+            sessionEventsManager = sessionEventsManager,
```

Comment:
SessionEventsManager was removed in iOS from here

Reply:
Cleaned it up 👍